src/braket/circuits/circuit.py

```diff
@@ -1043,25 +1045,93 @@ def diagram(self, circuit_diagram_class=AsciiCircuitDiagram) -> str:
         """
         return circuit_diagram_class.build_diagram(self)
 
-    def to_ir(self) -> Program:
+    def to_ir(
+        self,
+        ir_type: IRType = IRType.JAQCD,
+        qubit_reference_type: QubitReferenceType = QubitReferenceType.VIRTUAL,
+    ) -> Union[OpenQasmProgram, JaqcdProgram]:
         """
         Converts the circuit into the canonical intermediate representation.
         If the circuit is sent over the wire, this method is called before it is sent.
 
+        Args:
+            ir_type (IRType): The IRType to use for converting the circuit object to its
+                IR representation.
+            qubit_reference_type (QubitReferenceType): The qubit reference type to use for
+                the IR representation.
```

Comment:
Is setting the qubit_reference_type as PHYSICAL exactly the same as setting it as VIRTUAL, when ir is JAQCD? If so, we should at least document that this value is only relevant for OpenQASM.